### DISCUSSION TITLE
how to plot partial effect of a factor/ordered variable

### DISCUSSION THREAD

**Discussion by wymin576** [kogalur/randomForestSRC#427] [Q&A]
As we known, in the package randomForestSRC, partial.rfsrc can acquire partial effect a variable in survival analysis. The Reference manual from CRAN give the samples to use partial.rfsrc. however the two samples use numeric variable  to plot partial effect. i run this sample with factor/ordered variable and can not get result. so, anyone can give a sample with R?  thanks in advance.

**Comment by ishwaran** [comment]
Can you provide a reproducible example?

**Comment by wymin576** [comment]
rm(list = ls())
library(randomForestSRC)
library(ggRandomForests)
# if the variable is categorical
obj <- rfsrc(Surv(time,status)~., veteran)
gg_v <- gg_variable(obj)

plot(gg_v, xvar="celltype")

# if categorical variable is ordinal 
# the code can not work
veteran$celltype <- as.ordered(veteran$celltype)
v.obj <- rfsrc(Surv(time,status)~., veteran)
gg_v2 <- gg_variable(v.obj)

plot(gg_v2, xvar="celltype")

**Comment by ishwaran** [comment]
`ggRandomForests` is an external package developed by another group which is where the error is occurring.

Here's how to do partial plots entirely within the `randomForestSRC` package

```
data(veteran, package = "randomForestSRC")
v.obj <- rfsrc(Surv(time, status) ~ ., data = veteran)
plot.variable(v.obj,xvar.names="celltype",partial=TRUE)

```

**Reply by wymin576** [reply]
it is great! the code can run. thank you very much.
